# Phylogenetic tree building

## Input data

If you do not have your own data, please download example files, which include:<br/>
- [`alignment.fas`](https://raw.githubusercontent.com/Mycology-Microbiology-Center/Phylo2021/main/data/alignment.fas), aligned sequences in FASTA format;<br/>
- [`partitions.txt`](https://raw.githubusercontent.com/Mycology-Microbiology-Center/Phylo2021/main/data/partitions.txt), ...<br/>
- [`alignment_mrbayes.nex`](https://raw.githubusercontent.com/Mycology-Microbiology-Center/Phylo2021/main/data/alignment_mrbayes.nex), ...<br/>


You can download the files directly on a cluster with `wget` using the following syntax:
```bash
wget https://path/to/file
```


## RaxMl-NG

For maximum likelihood-based phylogenetic analysis we will use `RaxMl-NG`.<br/>

First, download the required input files:<br/>
```bash
wget https://raw.githubusercontent.com/Mycology-Microbiology-Center/Phylo2021/main/data/alignment.fas
wget https://raw.githubusercontent.com/Mycology-Microbiology-Center/Phylo2021/main/data/partitions.txt
```

Then, lest's create a script (`raxml.sh`) that will perform the following steps:<br/>

-
-
-
-


```bash
cat > raxml.sh <<'EOT'
#!/bin/bash

  ## Input data:
  # $1 = input fasta, e.g. alignment.fas
  # $2 = part model,  e.g. partitions.txt
  # $3 = prefix (T1)
  # $4 = prefix (T2)
  # $5 = prefix (T3)
  # $6 = prefix (T4)

echo "Start analysis for " "$1" " + " "$2"

# Sanity check of alignment
raxml-ng --check --msa "$1" --model "$2"

# ML search for best topology among 10 random + 10 parsimony trees
raxml-ng --msa "$1" --model "$2" --prefix "$3" --threads 4 --seed 123 --brlen scaled 

# Make 1000 classic bootstraps
raxml-ng --bootstrap --msa "$1" --model "$2" --prefix "$4" --seed 123 --threads 2 --bs-trees 1000

# Check BS convergence
raxml-ng --bsconverge --bs-trees "$4".raxml.bootstraps --prefix "$5" --seed 123 --threads 4 --bs-cutoff 0.01

# Calculate and apply branch support to the best tree found in round T2 ("$4")
raxml-ng --support --tree "$3".raxml.bestTree --bs-trees "$4".raxml.bootstraps --prefix "$6" --threads 4

echo "RaxMl analysis finished: " "$1" " + " "$2"
EOT
```
Please consider this script as a simplified example.<br/>
Here we use positional arguments (e.g., `$1`) for input data and parameters.<br/>
To keep the things simple, we also omit some saninty checks of the input data and do not trace the execution status of each command (if something is wrong with the data, there is a chance that commands could terminate with an error!).


To schedule the job use:
```bash
sbatch \
  --job-name=raxml \
  --ntasks-per-node=4 --nodes=1 --mem=10G \
  --time=02:00:00 \
  raxml.sh alignment.fas partitions.txt T1 T2 T3 T4
```



## MrBayes

For Bayesian inference of phylogenetic trees we will use `MrBayes`.<br/>

Create a script (`mrbayes.txt`) with commands to execute `MrBayes`:
```
cat > mrbayes.txt <<'EOT'
begin mrbayes;
set autoclose=yes nowarnings=yes;
execute alignment.nex;

charset POS1= 1-738\3;
charset POS2= 2-738\3;
charset POS3= 3-738\3;

partition Codons=2:POS1 POS2, POS3;
set partition=Codons;

lset rates=gamma nst=mixed;
unlink revmat=(all) pinvar=(all) shape=(all) statefreq=(all);    
prset applyto=(all) ratepr=variable;

mcmc ngen=1000000 nruns=4 nchains=4 temp=0.200 swapfreq=1 nswaps=1 samplefreq=1000 mcmcdiagn=Yes minpartfreq=0.1 allchains=No relburnin=Yes burninfrac=0.25 stoprule=Yes starttree=random stopval=0.01 Savebrlens=Yes nperts=0 Ordertaxa=Yes filename=mcmcout;

sump relburnin=Yes burninfrac=0.25 nruns=4 outputname=sumpout;  

sumt relburnin=Yes burninfrac=0.25 nruns=4 ntrees=1 minpartfreq=0.1 contype=Halfcompat conformat=simple outputname=sumtout;

end;
EOT
```
Please note, that some example-specific values are hard-coded in this scipt. If you use your own data, adjust the script accordingly.


Create a batch script:
```bash
cat > mrbayes.sh <<'EOT'
#!/bin/bash -l
#SBATCH --job-name=mrbayes
#SBATCH --cpus-per-task=4
#SBATCH --nodes=1
#SBATCH --mem=10G
#SBATCH --time=02:00:00

echo "Starting Mrbayes analysis"
mpirun -np 4 mb mrbayes.txt
echo "Analysis finished"
EOT
```



Schedule the job:
```bash
sbatch mrbayes.sh
```

