# Examples of input files
To download, right-click on the file names below and select "Save Link as...". All files can be also accessed from [here](https://github.com/Mycology-Microbiology-Center/Phylo2021/tree/main/data).

## RaxMl input
- [`alignment.fas`](https://raw.githubusercontent.com/Mycology-Microbiology-Center/Phylo2021/main/data/alignment.fas), aligned sequences in FASTA format;<br/>
- [`partitions.txt`](https://raw.githubusercontent.com/Mycology-Microbiology-Center/Phylo2021/main/data/partitions.txt), alignment coordinates of different parts of the gene (will be used as `RaxMl` input).<br/>

## MrBayes input
- [`alignment_mrbayes.nex`](https://raw.githubusercontent.com/Mycology-Microbiology-Center/Phylo2021/main/data/alignment_mrbayes.nex), aligned sequences in NEXUS format with additional `MrBayes` block containing commands for the analysis.<br/>


# Examples of resulting trees

- [`tree_raxml.nwk`](https://raw.githubusercontent.com/Mycology-Microbiology-Center/Phylo2021/main/data/tree_raxml.nwk), phylogenetic tree in Newick format constructed by RaxMl;<br/>
- [`tree_mrbayes.nex`](https://raw.githubusercontent.com/Mycology-Microbiology-Center/Phylo2021/main/data/tree_mrbayes.nex), phylogenetic tree in NEXUS format constructed by MrBayes;<br/>
- [`tree_mrbayes_figtree.nex`](https://raw.githubusercontent.com/Mycology-Microbiology-Center/Phylo2021/main/data/tree_mrbayes_figtree.nex), phylogenetic tree in NEXUS format constructed by MrBayes, in a FigTree-friendly format - only for demonstrational purposes.<br/>


# Input files used for visualization
- [`labels_simple.csv`](https://raw.githubusercontent.com/Mycology-Microbiology-Center/Phylo2021/main/data/labels_simple.csv), main table mapping tree labels to the publication-ready labels;<br/>
- [`labels_extended.csv`](https://raw.githubusercontent.com/Mycology-Microbiology-Center/Phylo2021/main/data/labels_extended.csv), alternative table mapping tree labels to the data needed to construct publication-ready labels.<br/>

[Home](index.md)




